### PR TITLE
feature/node-support

### PR DESCRIPTION
**Pull request by phyzical**
Fixes #22 

## Proposed Changes

updated cli context to have separate file and use node 
changed index.js to allow node usage
constructor to allow changing of arguments in node usage
added a run function for node usage

first pr separation from 

https://github.com/arfeo/npm-add-dependencies/pull/22

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,3 +46,23 @@ or with `npx`:
...
+const npmAdd = require('add-dependencies');
+//OR
+// Note: to use import you must have `"type": "module"` in your projects package.json
+import npmAdd from 'add-dependencies';
+const dependencies = [
+    'package1',
+    'package2',
+    'package3',
+];
+const target = npmAdd.CONSTANTS.DEPENDENCIES;
```
Comment (arfeo) [inline README.md:70]: What is `npmAdd.CONSTANTS` here? Can't see any object like this over here.

File: README.md [hunk 2]
```diff
@@ -46,3 +46,23 @@ or with `npx`:
 ```sh
 $ npx add-dependencies /home/user/project/package.json moment@2.0.0 react@16.8 redux eslint --dev
 ```
+
+or via nodejs
+
+```js
+const npmAdd = require('add-dependencies');
```
Comment (arfeo) [inline README.md:53]: You'd better describe imports separately. People usually just copy-paste example code snippets, and this line and line 62 will definitely lead to an error.

File: lib/AddDependencies.js [hunk 3]
```diff
@@ -59,7 +64,19 @@ class AddDependencies {
...
+    return this.addDependencies().then(() => this.saveToPackage());
+  }
+
   runNpmShow(dep) {
+    if (!dep) {
+      console.error(
+        '\x1b[31m%s\x1b[0m',
+        `No Dependency Provided to runNpmShow()`
+      );
+      return new Promise((resolve) => resolve());
```
Comment (arfeo) [inline lib/AddDependencies.js:78]: Please, add a blank line before the `return` statement.

File: README.md (hunk 1 above)
Comment (phyzical) [reply README.md:70]: whoops missed this in the separation

**Comment by phyzical** [comment]
hey @arfeo comments should be addressed, let me know if there is anything else.

**Pull request closed without merging**